### PR TITLE
Create a device

### PR DESCRIPTION
**Pull request by howbazaar**
Based on the devices support in the controller PR #20.

### CODE REVIEW

File: controller.go [hunk 1]
```diff
@@ -365,6 +404,20 @@ func (c *controller) post(path, op string, params url.Values) (interface{}, erro
 	return parsed, nil
 }
 
+func (c *controller) delete(path string) error {
+	path = EnsureTrailingSlash(path)
+	requestID := nextRequestID()
+	logger.Tracef("request %x: DELTE %s%s", requestID, c.client.APIURL, path)
```
Comment (mjs) [inline controller.go]: DELETE?

File: device.go [hunk 2]
```diff
@@ -62,8 +106,7 @@ func readDevices(controllerVersion version.Number, source interface{}) ([]*devic
 	if deserialisationVersion == version.Zero {
 		return nil, NewUnsupportedVersionError("no device read func for version %s", controllerVersion)
 	}
-	readFunc := deviceDeserializationFuncs[deserialisationVersion]
-	return readDeviceList(valid, readFunc)
+	return deviceDeserializationFuncs[deserialisationVersion], nil
```
Comment (mjs) [inline device.go]: Is there always going to be a matching entry in the map?

**Comment by mjs** [comment]
LGTM

File: controller.go (hunk 1 above)
Comment (howbazaar) [inline controller.go]: Fixed.

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**